tests/010-alpns.cpp

```diff
@@ -0,0 +1,111 @@
+#include <oxenc/hex.h>
+
+#include <catch2/catch_test_macros.hpp>
+#include <quic.hpp>
+#include <quic/gnutls_crypto.hpp>
+#include <thread>
+
+#include "utils.hpp"
+
+namespace oxen::quic::test
+{
+    using namespace std::literals;
+
+    TEST_CASE("010 - ALPNs", "[010][alpns][execute]")
+    {
+        auto server_tls = GNUTLSCreds::make("./serverkey.pem"s, "./servercert.pem"s, "./clientcert.pem"s);
+        auto client_tls = GNUTLSCreds::make("./clientkey.pem"s, "./clientcert.pem"s, "./servercert.pem"s);
+
+        opt::local_addr server_local{};
+        opt::local_addr client_local{};
+        opt::handshake_timeout timeout{500ms};
+
+        auto client_established = callback_waiter{[](connection_interface&) {}};
+
+        // this has to destroy *after* network, in case it doesn't go off before then
+        auto client_closed = callback_waiter{[](connection_interface&, uint64_t) {}};
+
+        Network test_net{};
+
+        SECTION("No ALPNs specified")
+        {
+            auto server_endpoint = test_net.endpoint(server_local, timeout);
+            REQUIRE(server_endpoint->listen(server_tls));
```

Comment:
Thought: I think there's a strong argument to make a lot of our REQUIRE's into CHECK's. It makes sense to me to save REQUIRE for critical targeted functionalities of the unit tests themselves